libbeat/publisher/queue/queue.go

```diff
@@ -34,6 +35,23 @@ type ACKListener interface {
 	OnACK(eventCount int) // number of consecutively published events acked by producers
 }
 
+//Metrics is a set of basic-user friendly metrics that report the current state of the queue. These metrics are meant to be relatively generic and high-level, and when reported directly, can be comprehensible to a user.
+type Metrics struct {
+	//EventCount is the total events currently in the queue
+	EventCount opt.Uint
+	//ByteCount is the total byte size of the queue
+	ByteCount opt.Uint
+	//ByteLimit is the user-configured byte limit of the queue
+	ByteLimit opt.Uint
+	//EventLimit is the user-configured event limit of the queue
+	EventLimit opt.Uint
+
+	//LongestWaitingItem is the timestamp of the oldest item in the queue.
+	LongestWaitingItem common.Time
+	//QueueLag is the difference between the consumer and producer position in the queue.
```

Comment:
What does "position" mean here? If it means byte ( / event) index in the queue buffer then this is very similar to `ByteCount` / `EventCount` (maybe modulo some rounding depending on how e.g. the disk queue reports events that are acknowledged but not yet deleted).

I'd rather leave this one off for now. I'd like things in the global interface to have an unambiguous meaning that would be consistently derivable for any queue, at least in principle, and I don't think this one is yet. If we could define it in language that is readily convertable to a unit test I'd feel better :-)

Reply:
So, my understanding is that this will be a measure of the "delay" involved in an event entering then exiting a queue, as told by the event IDs. Since the IDs increment, if the event leaving the queue for the output has `ID = 10` and the event entering has `ID =15`, the QueueLag is 5.

Reply:
Hmm, but if the event leaving the queue has id 10 and the event entering has id 15, then isn't that also the same as `EventCount` being 5? (+- 1 anyway) It seems redundant with that definition.

Reply:
Yah, I'm probably describing it wrong, and discussion of this is spread out across like, 3 different issues, which isn't helping. From Craig's earlier comment:

> queue lag is just the difference between the sequence numbers of the newest and oldest item in the queue.

Reply:
I don't know, to me that still sounds the same? I don't see how it isn't redundant with `EventCount`. Is there any situation where this should be different than "the number of items in the queue"?